uvicorn/protocols/websockets/wsproto_impl.py

```diff
@@ -255,10 +255,8 @@ async def send(self, message):
                 )
                 self.handshake_complete = True
                 self.close_sent = True
-                msg = h11.Response(status_code=403, headers=[])
+                msg = events.RejectConnection(status_code=403, headers=[])
```

Comment:
Yup, looks good to me — results in the desired behavior, same as the `websockets` case.